### ISSUE TITLE
Some words about Tentacle NPC collision bug

### ISSUE THREAD

**Issue by TheMostUpset** [Facepunch/garrysmod-issues#3460]
So, as we know, Tentacle NPC is broken and doesn't do any damage to player. But here are some things I've noticed:
Tentacles are unsolid after spawn and have small collision box at the "root". Same thing happens on HL:S maps, BUT when scientist scripted scene is completed, one of them actually becames solid and does damage.
https://puu.sh/AgP5z/29c0fbc9f9.webm

I really hope it helps somehow.

**Comment by robotboy655** [comment]
There is a problem with animations serverside which makes the collision meshes not follow the model as it appears clientside. You can use vcollide_wireframe to see it.

**Comment by TheMostUpset** [comment]
Yeah, but what happpened after scripted scene?

**Comment by Kefta** [comment]
I believe the issue with tentacles is that they don't check their physics contacts until they are in ``NPC_STATE_SCRIPT`` since ``CNPC_Tentacle::Cycle`` is the initial Think function as opposed to ``CAI_BaseNPC::NPCThink``. This can be fixed by adding this code to ``CNPC_Tentacle::Cycle`` below the ``NPC_STATE_SCRIPT`` check but above ``StudioFrameAdvance();``:

```
if ( m_bCheckContacts )
{
	CheckPhysicsContacts();
}
```

**Comment by robotboy655** [comment]
Nope, that's not it. I have also tried just calling NPCThink from Cycle, but that doesn't fix it either. I also tried looking for differences in the save table and that didn't yield any results either.

**Comment by Kefta** [comment]
It must be something with ``NPC_STATE_SCRIPT`` because running ``Tentacle:SetNPCState(NPC_STATE_SCRIPT)`` causes the hitbox to start following the entity.

**Comment by robotboy655** [comment]
I think the only reason that works is because of whatever process that resets the state back to idle.

**Comment by Kefta** [comment]
Found a proper fix (the current one of always forcing the PVS can still mess up under certain conditions). Add ``CheckPVSCondition();`` to ``Cycle`` for the condition to automatically update.
